lib/idx/remediators/Base/SelectAuthenticator.ts

```diff
@@ -59,6 +59,14 @@ export class SelectAuthenticator<T extends SelectAuthenticatorValues = SelectAut
     // Proceed with provided authenticators
     const matchedOption = this.findMatchedOption(authenticators, options!);
     if (matchedOption) {
+
+      // fix for OKTA-612939 (below) seems to have caused a bug when trying to re-select authenticators
+      // with multiple methodTypes. If `options.step` is passed, this remediation is explicitly being
+      // invoked, therefore do not guard against auto-remediating the selected authenticator (OKTA-646147)
+      if (this.options.step) {
```

Comment:
```suggestion
      if (this.options.step || this.values.methodType) {
```
What do you think about this fix?
Should cover case if user doesn't provide `step`, eg:
```js
await oktaAuth.idx.authenticate({ username: '...', password: '...' })
await oktaAuth.idx.proceed({ authenticator: 'phone_number', methodType: 'sms' })
await oktaAuth.idx.proceed({ authenticator: 'phone_number', methodType: 'voice' })
```